receiver/mongodbatlasreceiver/logs.go

```diff
@@ -0,0 +1,337 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mongodbatlasreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbatlasreceiver"
+
+import (
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"go.mongodb.org/atlas/mongodbatlas"
+	"go.opentelemetry.io/collector/component"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbatlasreceiver/internal/model"
+)
+
+// combindedLogsReceiver wraps alerts and log receivers in a single log receiver to be consumed by the factory
+type combindedLogsReceiver struct {
+	alerts *alertsReceiver
+	logs   *receiver
+}
+
+type resourceInfo struct {
+	Org      *mongodbatlas.Organization
+	Project  *mongodbatlas.Project
+	Cluster  *mongodbatlas.Cluster
+	Hostname string
+	LogName  string
+}
+
+// MongoDB Atlas Documentation reccommends a polling interval of 5  minutes: https://www.mongodb.com/docs/atlas/reference/api/logs/#logs
+const collection_interval = time.Minute * 5
+
+// Starts up the combined MongoDB Atlas Logs and Alert Receiver
+func (c *combindedLogsReceiver) Start(ctx context.Context, host component.Host) error {
+	var errs error
+
+	// If we have an alerts receiver start alerts collection
+	if c.alerts != nil {
+		if err := c.alerts.Start(ctx, host); err != nil {
+			errs = multierror.Append(errs, err)
+		}
+	}
+
+	// If we have a logging receiver start log collection
+	if c.logs != nil {
+		if err := c.logs.Start(ctx, host); err != nil {
+			errs = multierror.Append(errs, err)
+		}
+	}
+
+	return errs
+}
+
+// Shuts Down the Alerts and Log combined receiver
```

Comment:
really nitpicky: but i feel like the wording on this comment should match the wodring on the Start comment